src/FileIO.pm

```diff
@@ -112,7 +137,45 @@ sub WriteFile
 
   if(open(my $fh, '>', $file)) {
     print $fh $l;
-    close $fh;
+    if(!close($fh)) {
+      $self->error("Failed to close $file: $!");
+      $ok = 0;
+    }
+  }
+  else {
+    $self->error("Failed to open $file: $!");
+    $ok = 0;
+  }
+
+  umask $saved_umask;
```

Comment:
No, the umask fix was in Core.pm.

This is basically a port of the changes from sle11 and aligns both code streams.